apidef/oas/schema/x-tyk-api-gateway.json

```diff
@@ -0,0 +1,915 @@
+        },
+        "header": {
+          "$ref": "#/definitions/X-Tyk-AuthSource"
+        },
+        "cookie": {
+          "$ref": "#/definitions/X-Tyk-AuthSource"
+        },
+        "query": {
+          "$ref": "#/definitions/X-Tyk-AuthSource"
+        },
+        "allowedAlgorithms": {
+          "type": "array",
+          "items": [
+            {
+              "type": "string"
+            }
+          ]
+        },
+        "allowedClockSkew": {
+          "type": "number",
+          "format": "double"
+        }
+      },
+      "required": [
+        "enabled"
+      ]
+    },
+    "X-Tyk-OIDC": {
+      "type": "object",
+      "properties": {
+        "enabled": {
+          "type": "boolean"
+        },
+        "header": {
+          "$ref": "#/definitions/X-Tyk-AuthSource"
+        },
+        "cookie": {
+          "$ref": "#/definitions/X-Tyk-AuthSource"
+        },
+        "query": {
+          "$ref": "#/definitions/X-Tyk-AuthSource"
+        },
+        "segregateByClientId": {
+          "type": "boolean"
+        },
+        "providers": {
+          "type": "array",
+          "items": [
+            {
+              "$ref": "#/definitions/X-Tyk-Provider"
+            }
+          ]
+        },
+        "scopes": {
+          "$ref": "#/definitions/X-Tyk-Scopes"
+        }
+      },
+      "required": [
+        "enabled"
+      ]
+    },
+    "X-Tyk-CustomPlugin": {
+      "type": "object",
+      "properties": {
+        "enabled": {
+          "type": "boolean"
+        },
+        "header": {
+          "$ref": "#/definitions/X-Tyk-AuthSource"
+        },
+        "cookie": {
+          "$ref": "#/definitions/X-Tyk-AuthSource"
+        },
+        "query": {
+          "$ref": "#/definitions/X-Tyk-AuthSource"
+        }
+      },
+      "required": [
+        "enabled"
+      ]
+    },
+    "X-Tyk-GoPlugin": {
+      "type": "object",
+      "properties": {
+        "enabled": {
+          "type": "boolean"
+        }
+      },
+      "required": [
+        "enabled"
+      ]
+    },
+    "X-Tyk-Authentication": {
+      "type": "object",
+      "properties": {
+        "enabled": {
+          "type": "boolean"
+        },
+        "stripAuthorizationData": {
+          "type": "boolean"
+        },
+        "baseIdentityProvider": {
+          "type": "string",
+          "enum": [
+            "auth_token",
+            "hmac_key",
+            "basic_auth_user",
+            "jwt_claim",
+            "oidc_user",
+            "oauth_key",
+            ""
+          ]
+        },
+        "hmac": {
+          "$ref": "#/definitions/X-Tyk-HMAC"
+        },
+        "oidc": {
+          "$ref": "#/definitions/X-Tyk-OIDC"
+        },
+        "goPlugin": {
+          "$ref": "#/definitions/X-Tyk-GoPlugin"
+        },
+        "customPlugin": {
+          "$ref": "#/definitions/X-Tyk-CustomPlugin"
+        },
+        "securitySchemes": {
+          "type": "object",
+          "^\\/": {
+            "anyOf": [
```

Comment:
shouldn't it be `oneOf`? Native one uses that.

Reply:
updated